drivers/storage/portworx/portworx.go

```diff
@@ -153,7 +155,176 @@ func (p *portworx) GetSelectorLabels() map[string]string {
 	return pxutil.SelectorLabels()
 }
 
-func (p *portworx) SetDefaultsOnStorageCluster(toUpdate *corev1.StorageCluster) {
+func (p *portworx) setMaxStorageNodePerZone(toUpdate *corev1.StorageCluster, upgradingPortworx bool) error {
+	// Set max node per zone
+	// if no value is set for any of max_storage_nodes*, try to see if can set a default value
+	if toUpdate.Spec.CloudStorage != nil &&
+		toUpdate.Spec.CloudStorage.MaxStorageNodesPerZonePerNodeGroup == nil &&
+		toUpdate.Spec.CloudStorage.MaxStorageNodesPerZone == nil &&
+		toUpdate.Spec.CloudStorage.MaxStorageNodes == nil {
+		maxStorageNodesPerZone := uint32(0)
+		var err error
+
+		nodeList := &v1.NodeList{}
+		err = p.k8sClient.List(context.TODO(), nodeList, &client.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		if pxutil.IsFreshInstall(toUpdate) {
+			// Let's do this only when it's a fresh install of px
+			// TODO:  get node list
```

Comment:
Is this a TODO for this PR, or later? If later, is there a PWX?

Reply:
oh sorry, already resolved in this PR, will delete this.